braindecode/preprocessing/preprocess.py

```diff
@@ -42,7 +42,9 @@ class Preprocessor(object):
     kwargs:
         Keyword arguments to be forwarded to the MNE function.
     """
-    def __init__(self, fn, apply_on_array=True, **kwargs):
+    def __init__(self, fn, *, apply_on_array=True, **kwargs):
+        if hasattr(fn, '__name__') and fn.__name__ == '<lambda>':
+            raise ValueError('lambda function not supported.')
```

Comment:
Could we give warning instead that the preprocessing then will not be serializable? I am afraid we will make API hard to learn by more and more (hidden) restrictions.

Reply:
> > @hubertjb I just had to learn that we cannot use `numpy.multiply` with our current `Preprocessor`. The reason is that `numpy.multiply` requires two **positional** arguments (it actually forces two positional arguments using `/`, see https://numpy.org/doc/stable/reference/generated/numpy.multiply.html, https://www.python.org/dev/peps/pep-0570/) where our `Preprocessor` only supports keyword arguments (https://github.com/braindecode/braindecode/blob/master/braindecode/preprocessing/preprocess.py#L45). Maybe it is worth keeping `braindecode.preprocessing.scale`.
> 
> Interesting, I was not aware of pep570! And sorry about that, I should have been more careful. Let's maybe reinstate it but rename it `braindecode.preprocessing.multiply`, to avoid confusion with sklearn's `scale` and `robust_scale`?

Ah, no worries. I also was not aware. Yes we should avoid confusion with sklearn, although I am very unhappy about their choices here. Both with forcing positional arguments as well as with naming 'scale'. I think `standard_scale` would be much more appropriate.

Reply:
We could. If the user insists on using a lambda function it will show up as
`{'<lambda>': {}}` in the lists below.